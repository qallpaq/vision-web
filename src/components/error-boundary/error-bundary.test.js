import React from 'react'
import ErrorBoundary from "./error-boundary"
import {mount} from "enzyme"


const Child = () => {
    return (
        <div className='child'>
            child
        </div>
    )
}

describe('ErrorBoundary', () => {
    const wrapper = mount(<ErrorBoundary children={<Child/>}/>)

    it('should return snapshot', () => {
        expect(wrapper).toMatchSnapshot()
    })

    it('should contain child', () => {
        const result = wrapper.find('.child')
        expect(result.text()).toBe('child')
    })

    it('should return ErrorIndicator', () => {
        wrapper.setState({ hasError: true })
        const result = wrapper.find('h1')
        expect(result.text()).toBe('Something goes wrong...')
    })
})
