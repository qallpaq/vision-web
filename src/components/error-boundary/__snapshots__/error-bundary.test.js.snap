// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary should return snapshot 1`] = `
<ErrorBoundary>
  <Child>
    <div
      className="child"
    >
      child
    </div>
  </Child>
</ErrorBoundary>
`;
